apps/meteor/ee/app/livechat-enterprise/server/api/lib/departments.ts

```diff
@@ -0,0 +1,36 @@
+import { LivechatDepartmentAgents, LivechatUnit } from '@rocket.chat/models';
+
+export const getDepartmentsWhichUserCanAccess = async (userId: string): Promise<string[]> => {
+	const departments = await LivechatDepartmentAgents.find(
+		{
+			agentId: userId,
+		},
+		{
+			projection: {
+				departmentId: 1,
+			},
+		},
+	).toArray();
+
+	const monitoredDepartments = await LivechatUnit.findMonitoredDepartmentsByMonitorId(userId);
+	const combinedDepartments = [
+		...departments.map((department) => department.departmentId),
+		...monitoredDepartments.map((department) => department._id),
+	];
+
+	return [...new Set(combinedDepartments)];
+};
+
+export const hasAccessToDepartment = async (userId: string, departmentId: string): Promise<boolean> => {
+	const department = await LivechatDepartmentAgents.findOneByAgentIdAndDepartmentId(userId, departmentId);
+	if (department) {
+		return true;
+	}
+
+	const monitoredDepartments = await LivechatUnit.findMonitoredDepartmentsByMonitorId(userId);
```

Comment:
You can create a new query to check the department is on the scope for the monitor instead of fetching all the ones it can view 👀

Reply:
Good idea! This should be implemented now. Pls check